modules/programs/mu.nix

```diff
@@ -0,0 +1,86 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.mu;
+
+  # By default, mu will place it's database in XDG_CACHE_HOME/mu.
+  defaultDBLocation = config.xdg.cacheHome + "/mu";
+
+  # Used to generate command line arguments that mu can operate with.
+  genCmdMaildir = path: "--maildir=" + path;
+
+  genCmdAddrs = addrs:
+    let
+      # Takes the list of address strings and returns list of strings with the
+      # --my-address= prefixed to each
+      addMyAddr = addrs: builtins.map (addr: "--my-address=" + addr) addrs;
+    in builtins.concatStringsSep " " (addMyAddr addrs);
+
+in {
+  meta.maintainers = [ maintainers.KarlJoad ];
+
+  options.programs.mu = {
+    enable = mkEnableOption "mu, a maildir indexer and searcher";
+    # No options/config file present for mu, and program author will not be adding
+    # one soon.
+    # See https://github.com/djcb/mu/issues/882 for more information about this.
+
+    # All configuration options below affect only the initial index script wrapper.
+
+    maildir = mkOption {
```

Comment:
I would suggest removing this option and just unconditionally use the `config.accounts.email.maildirBasePath` value.

Reply:
That's fine. I just wanted to present the option to move the `maildir` option because `mu` doesn't require the directory where mail is stored to be any default (in which case, the path must be given each time).

It is cleared up now.